clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -470,7 +445,7 @@ public static AbstractResponse parseResponse(ByteBuffer responseBuffer, RequestH
      */
     private void processDisconnection(List<ClientResponse> responses, String nodeId, long now) {
         connectionStates.disconnected(nodeId, now);
-        nodeApiVersions.remove(nodeId);
+        nodeApiVersions.remove(Integer.valueOf(nodeId));
```

Comment:
I was thinking about this a little more. I wonder if we could just change the key type to string to make it consistent with `nodesNeedingApiVersionsFetch`?